Fix org.openrewrite.java.jackson.UpgradeJackson_2_3 - Jackson3 no longer uses jackson-datatype-jsr310. Recipe should remove references to it

<!--
Thank you for reporting an issue with OpenRewrite!
We appreciate you taking the time to help us improve.
Please fill out the template below to help us understand and reproduce the issue.
Feel free to delete any sections that don't apply to your issue.
-->

## What version of OpenRewrite are you using?
<!--
Whenever possible please try to replicate your issue with the latest versions of OpenRewrite.
The latest major and minor versions of OpenRewrite projects are usually listed here:
https://docs.openrewrite.org/reference/latest-versions-of-every-openrewrite-module
For patch releases check the GitHub Releases page for the respective project.

We release every few weeks, so it's possible that your issue has already been fixed.

If you want to try the most recent changes that haven't been fully released yet, you can check out our snapshot releases
https://docs.openrewrite.org/reference/snapshot-instructions
-->
I am using

Using active recipe(s) [org.openrewrite.java.spring.boot4.UpgradeSpringBoot_4_0]

## How are you running OpenRewrite?
Executing maven commmand line

## What is the smallest, simplest way to reproduce the problem?
<!--
Sometimes the logs indicate a recipe stumbled over a particular pattern of code.
If you can share a code snippet that reproduces the issue, that will help us fix it faster.
We also accept [pull requests that merely replicate an issue](https://github.com/openrewrite/.github/blob/main/CONTRIBUTING.md#contributing-fixes), as a step up to a full fix.

A code snippet can be something simple like this, or similar for other languages:
-->
```java
JavaTimeModule module = new JavaTimeModule();
        objectMapper = new ObjectMapper();
        objectMapper.registerModule(module);
```

## What did you expect to see?
<!-- A code snippet, or a description of the behavior you expected helps us write a test to ensure the issue is fixed. -->
```java
        objectMapper = new ObjectMapper();
```

## What did you see instead?
<!-- A code snippet, or a description of the behavior you saw instead of the above expected result. -->
```java
//not removed
JavaTimeModule module = new JavaTimeModule();
        objectMapper = new ObjectMapper();
        objectMapper.registerModule(module);
```

## What is the full stack trace of any errors you encountered?
<!-- When errors occur, please include the output of `--stacktrace` for Gradle or `--debug` for Maven. -->
```
[ERROR]   symbol:   method registerModule(com.fasterxml.jackson.datatype.jsr310.JavaTimeModule)
[ERROR]   location: variable objectMapper of type tools.jackson.databind.ObjectMapper
```

## Are you interested in [contributing a fix to OpenRewrite](https://github.com/openrewrite/.github/blob/main/CONTRIBUTING.md#contributing-fixes)?
<!-- Indicate if this is something you would like to work on, and how we can best support you in doing so. -->
Update recipe to include removing any reference to jackson-datatype-jsr310

Integrated Functionality: The logic previously contained in the external jackson-datatype-jsr310 module has been merged into the main jackson-databind package.
No Module Registration Required: You do not need to explicitly register a JavaTimeModule or JSR310Module with the ObjectMapper in Jackson 3.0; the support works automatically by default.

--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -13,6 +13,7 @@ dependencies {
     implementation("org.openrewrite:rewrite-properties")
     implementation("org.openrewrite:rewrite-yaml")
     implementation("org.openrewrite.recipe:rewrite-java-dependencies:$rewriteVersion")
+    implementation("org.openrewrite.recipe:rewrite-static-analysis:${rewriteVersion}")
     implementation("org.assertj:assertj-core:latest.release")
 
     annotationProcessor("org.openrewrite:rewrite-templating:$rewriteVersion")

--- a/src/main/java/org/openrewrite/java/jackson/RemoveBuiltInModuleRegistrations.java
+++ b/src/main/java/org/openrewrite/java/jackson/RemoveBuiltInModuleRegistrations.java
@@ -80,6 +80,15 @@ public class RemoveBuiltInModuleRegistrations extends Recipe {
                         return super.visitMethodInvocation(method, ctx);
                     }
 
+                    @Override
+                    public @Nullable J visitVariableDeclarations(J.VariableDeclarations multiVariable, ExecutionContext ctx) {
+                        J.VariableDeclarations mv = (J.VariableDeclarations) super.visitVariableDeclarations(multiVariable, ctx);
+                        if (BUILT_IN_MODULES.contains(TypeUtils.toString(mv.getType()))) {
+                            return null;
+                        }
+                        return mv;
+                    }
+
                     private boolean isBuiltInModuleInstantiation(Expression expr) {
                         if (expr instanceof J.NewClass) {
                             J.NewClass newClass = (J.NewClass) expr;
@@ -87,7 +96,7 @@ public class RemoveBuiltInModuleRegistrations extends Recipe {
                                 return BUILT_IN_MODULES.contains(TypeUtils.toString(newClass.getClazz().getType()));
                             }
                         }
-                        return false;
+                        return BUILT_IN_MODULES.contains(TypeUtils.toString(expr.getType()));
                     }
                 }
         );

--- a/src/test/java/org/openrewrite/java/jackson/RemoveBuiltInModuleRegistrationsTest.java
+++ b/src/test/java/org/openrewrite/java/jackson/RemoveBuiltInModuleRegistrationsTest.java
@@ -280,4 +280,34 @@ class RemoveBuiltInModuleRegistrationsTest implements RewriteTest {
           )
         );
     }
+
+    @Issue("https://github.com/openrewrite/rewrite-jackson/issues/52")
+    @Test
+    void removeVariableDeclarations() {
+        rewriteRun(
+          java(
+            """
+              import com.fasterxml.jackson.databind.ObjectMapper;
+              import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+
+              class Test {
+                  void configure() {
+                      JavaTimeModule module = new JavaTimeModule();
+                      ObjectMapper mapper = new ObjectMapper();
+                      mapper.registerModule(module);
+                  }
+              }
+              """,
+            """
+              import com.fasterxml.jackson.databind.ObjectMapper;
+
+              class Test {
+                  void configure() {
+                      ObjectMapper mapper = new ObjectMapper();
+                  }
+              }
+              """
+          )
+        );
+    }
 }

